src/NumericalAlgorithms/SphericalHarmonics/Spherepack.cpp

```diff
@@ -892,12 +892,23 @@ void Spherepack::fill_scalar_work_arrays() {
 }
 
 DataVector Spherepack::prolong_or_restrict(const DataVector& spectral_coefs,
-                                           const Spherepack& target) const {
-  ASSERT(spectral_coefs.size() == spectral_size(),
-         "Expecting " << spectral_size() << ", got " << spectral_coefs.size());
-  DataVector result(target.spectral_size(), 0.0);
-  SpherepackIterator src_it(l_max_, m_max_);
-  SpherepackIterator dest_it(target.l_max_, target.m_max_);
+                                           const size_t l_max_coefs,
+                                           const size_t m_max_coefs,
+                                           const size_t l_max_target,
+                                           const size_t m_max_target) {
+  if (l_max_coefs == l_max_target and m_max_coefs == m_max_target) {
+    return spectral_coefs;
+  }
+
+  ASSERT(spectral_coefs.size() ==
+             Spherepack::spectral_size(l_max_coefs, m_max_coefs),
+         "Expecting " << Spherepack::spectral_size(l_max_coefs, m_max_coefs)
+                      << ", got " << spectral_coefs.size());
+  const size_t spectral_size_target =
+      Spherepack::spectral_size(l_max_target, m_max_target);
+  DataVector result{spectral_size_target, 0.0};
```

Comment:
I don' think you need to zero-init this because you set the values below. We've found zero-init has non-negligible overhead.